lightning/src/ln/peer_handler.rs

```diff
@@ -2157,6 +2158,19 @@ where
 				self.message_handler.onion_message_handler.peer_disconnected(their_node_id);
 				return Err(PeerHandleError {}.into());
 			}
+			let sends_handler = &self.message_handler.send_only_message_handler;
+			if let Err(()) = sends_handler.peer_connected(their_node_id, &msg, inbound) {
+				log_debug!(
+					logger,
+					"Sending-Only Message Handler decided we couldn't communicate with peer {}",
+					log_pubkey!(their_node_id)
+				);
+				self.message_handler.route_handler.peer_disconnected(their_node_id);
+				self.message_handler.chan_handler.peer_disconnected(their_node_id);
+				self.message_handler.onion_message_handler.peer_disconnected(their_node_id);
+				self.message_handler.custom_message_handler.peer_disconnected(their_node_id);
+				return Err(PeerHandleError {}.into());
+			}
```

Comment:
Hmm... I guess this is unrelated to this PR. But why does an error indicate we couldn't communicate with the peer? Couldn't is simply be an internal handling error? The docs say:

```rust
	/// May return an `Err(())` if the features the peer supports are not sufficient to communicate
	/// with us. Implementors should be somewhat conservative about doing so, however, as other
	/// message handlers may still wish to communicate with this peer.
```

But isn't that why we have `provided_init_features`?